src/algorithm/parseSolidAuthorizationHeader.ts

```diff
@@ -0,0 +1,38 @@
+import { JwtStructureError } from "../error";
+import type { SolidJwt } from "../type";
+import { verifyAuthenticationScheme } from "./verifyAuthenticationScheme";
+
+/**
+ * Parse an authorization header used in Solid
+ *
+ * Verify the authorization header uses a known authentication scheme;
+ * verify the token's structure and return an object representing the JWT
+ * structure and the authentication scheme.
+ *
+ * @param authorizationHeader The authorization header used for the request.
+ * @return {SolidJwt} A representation of the authorization header.
+ */
+export function parseSolidAuthorizationHeader(
+  authorizationHeader: string
+): SolidJwt {
+  // Verify the authentication scheme is supported
+  verifyAuthenticationScheme(authorizationHeader);
+
+  const structure = authorizationHeader
+    .replace(/^(DPoP|Bearer) +/i, "")
+    .split(".");
+
+  // Verify the JWT consists of three concatenated strings separated by dots
+  if (structure.length !== 3) {
+    throw new JwtStructureError(`${structure.length}`);
+  }
```

Comment:
I'd do that with `/^(?:DPoP|Bearer) +(\w+)\.(\w+)\.(\w+)$/i` so that we're sure in one go it has the right parts; the happy path can inline line 19 and skip the test on line 31.

Something like:
```javascript
const  [ match, isDPoP, joseHeader, jwsPayload, jwsSignature ] =
    /^(?:(DPoP)|Bearer) +(\w+)\.(\w+)\.(\w+)$/i.exec(authorizationHeader);
if (!match) {
  // if desired, further analyze header for detailed error reporting
}
```

Reply:
Good suggestion, I refactored, much cleaner now: https://github.com/solid/access-token-verifier/pull/20/commits/bbd98334ccefd3d7f86864df1a080fa576a49103